app/main.py

```diff
@@ -87,128 +1,20 @@
+from app.knights.character import Knight
+from app.battle import Battle
+from app.knights.knights_data import KNIGHTS
 
 
-def battle(knightsConfig):
-    # BATTLE PREPARATIONS:
+def battle(knights_config: dict) -> dict:
+    lancelot = Knight(**knights_config["lancelot"])
+    arthur = Knight(**knights_config["arthur"])
+    mordred = Knight(**knights_config["mordred"])
+    red_knight = Knight(**knights_config["red_knight"])
 
-    # lancelot
-    lancelot = knightsConfig["lancelot"]
+    battle1 = Battle(lancelot, mordred).simulate()
+    battle2 = Battle(arthur, red_knight).simulate()
```

Comment:
The creation of `Battle` objects and the subsequent call to `simulate()` could be refactored to avoid repetition. Consider using a loop or a helper function to handle these operations, which would make the code cleaner and adhere to the DRY principle.